c10/core/TensorImpl.h

```diff
@@ -1318,6 +1318,27 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     device_opt_ = storage_.device();
   }
 
+  virtual void update_strides_to_format(MemoryFormat memory_format) {
+    switch (memory_format) {
+      case MemoryFormat::Any:
+      case MemoryFormat::Contiguous: {
+        auto old_dim = sizes_.size();
+        update_to_contiguous_strides(old_dim);
+        return;
+      }
+      case MemoryFormat::ChannelsLast: {
+        TORCH_CHECK(
+            dim() == 4,
+            "required rank 4 tensor to use channels_last format");
+        set_sizes_and_strides(sizes(), get_channels_last_strides(sizes()));
+        refresh_contiguous();
+        return;
+      }
+      case MemoryFormat::Preserve:
+        TORCH_CHECK(false, "unsupported memory format")
```

Comment:
does TORCH_CHECK support outputting the format?  If not, can you add support?  If so, can you add it to the output?